packages/collaboration/src/browser/collaboration-frontend-contribution.ts

```diff
@@ -0,0 +1,305 @@
+                    if (roomClaim.loginToken) {
+                        localStorage.setItem(COLLABORATION_AUTH_TOKEN, roomClaim.loginToken);
+                    }
+                    this.currentInstance?.dispose();
+                    const connection = await authHandler.connect();
+                    this.currentInstance = this.collaborationInstanceFactory({
+                        role: 'host',
+                        connection
+                    });
+                    this.currentInstance.onDidClose(() => {
+                        this.setStatusBarEntryDefault();
+                    });
+                    const roomCode = roomClaim.roomToken;
+                    this.setStatusBarEntryShared(roomCode);
+                    this.displayCopyNotification(roomCode, true);
+                } catch (err) {
+                    await this.messageService.error(nls.localize('theia/collaboration/failedCreate', 'Failed to create room: {0}', err.message));
+                }
+            }
+        });
+        commands.registerCommand(CollaborationCommands.JOIN_ROOM, {
+            execute: async () => {
+                try {
+                    const authHandler = await this.authHandlerDeferred.promise;
+                    const id = await this.quickInputService?.input({
+                        placeHolder: nls.localize('theia/collaboration/enterCode', 'Enter collaboration session code')
+                    });
+                    if (!id) {
+                        return;
+                    }
+                    const roomClaim = await authHandler.joinRoom(id);
+                    if (roomClaim.loginToken) {
+                        localStorage.setItem(COLLABORATION_AUTH_TOKEN, roomClaim.loginToken);
+                    }
+                    this.currentInstance?.dispose();
+                    const connection = await authHandler.connect();
+                    this.currentInstance = this.collaborationInstanceFactory({
+                        role: 'guest',
+                        connection
+                    });
+                    this.currentInstance.onDidClose(() => {
+                        this.setStatusBarEntryDefault();
+                    });
+                    this.setStatusBarEntryConnected(roomClaim.roomToken);
+                    await this.currentInstance.initialize();
+                } catch (err) {
+                    await this.messageService.error(nls.localize('theia/collaboration/failedJoin', 'Failed to join room: {0}', err.message));
+                }
+            }
+        });
+    }
+
+    protected async displayCopyNotification(code: string, firstTime = false): Promise<void> {
+        navigator.clipboard.writeText(code);
+        const notification = nls.localize('theia/collaboration/copiedInvitation', 'Invitation code copied to clipboard.');
+        if (firstTime) {
+            const makeReadonly = nls.localize('theia/collaboration/makeReadonly', 'Make read-only');
```

Comment:
```suggestion
            const makeReadonly = nls.localize('theia/collaboration/makeReadonly', 'Disable Workspace Editing');
```

Reply:
FYI, I couldn't apply this change as `Disable Workspace Editing` is too long for notification buttons and always gets truncated.